### DISCUSSION TITLE
Lighthouse CI checks on PRs

### DISCUSSION THREAD

**Discussion by zarahzachz** [CMSgov/design-system#2187] [Ideas]
I stumbled across the Spotify engineering blog and they break down [how they create Web performance stories](https://engineering.atspotify.com/2022/09/from-development-to-real-users-how-to-create-a-web-performance-story/). 

One thing that caught my eye was their use of [Lighthouse CI](https://github.com/GoogleChrome/lighthouse-ci) to check PRs for performance changes and I was thinking something like this could be a good way to make sure our docs site (and our components, by extension) was performant.

Lighthouse also does a11y testing, so this could provide additional coverage in that space as well.

One downer is all the guidance I'm reading using Github Actions to implement their CI, so if we went this route, we'd have to figure out how to get this working with Jenkins.

**Comment by ghost** [comment]
I like the idea, the reports are really nice.

It does look possible to integrate via jenkins, since it's all using npm packages to do the work from what I can see.

Seems like a good discovery ticket at the least, to see if it's possible to integrate without too much trouble. We could run the CI against our storybook stories like our current A11y testing is doing.
